boat_simulator/nodes/physics_engine/fluid_forces.py

```diff
@@ -50,11 +82,73 @@ def compute(self, apparent_velocity: NDArray, orientation: Scalar) -> Tuple[NDAr
         Returns:
             Tuple[NDArray, NDArray]: A tuple containing the lift force and drag force experienced
                 by the medium, both expressed in newtons (N).
+                Bound to 360 degrees before using.
         """
 
-        # TODO: Implement this method.
+        attack_angle = self.calculate_attack_angle(apparent_velocity, orientation)
+        lift_coefficient, drag_coefficient, area = self.interpolate(attack_angle)
+        velocity_magnitude = np.linalg.norm(apparent_velocity)
+
+        # Calculate the lift and drag forces
+        lift_force_magnitude = (
+            0.5 * self.__fluid_density * lift_coefficient * area * (velocity_magnitude**2)
+        )
+
+        drag_force_magnitude = (
+            0.5 * self.__fluid_density * drag_coefficient * area * (velocity_magnitude**2)
+        )
+
+        # Rotate the lift and drag forces by 90 degrees to obtain the lift and drag forces
+        # Rotate counter clockwise in 1st and 3rd quadrant, and clockwise in 2nd and 4th quadrant
+        # 0 otherwise
+        drag_force_direction = (apparent_velocity) / velocity_magnitude
+
+        def rotate_vector_clockwise(v, theta_degrees):
+            theta_radians = np.deg2rad(theta_degrees)  # Convert angle from degrees to radians
+            rotation_matrix = np.array(
+                [
+                    [np.cos(theta_radians), np.sin(theta_radians)],
+                    [-np.sin(theta_radians), np.cos(theta_radians)],
+                ]
+            )
+            v_rotated = np.dot(rotation_matrix, v)  # Multiply the rotation matrix by the vector
+            return v_rotated
+
+        def rotate_vector_anticlockwise(v, theta_degrees):
+            theta_radians = np.deg2rad(theta_degrees)  # Convert angle from degrees to radians
+            rotation_matrix = np.array(
+                [
+                    [np.cos(theta_radians), -np.sin(theta_radians)],
+                    [np.sin(theta_radians), np.cos(theta_radians)],
+                ]
+            )
+            v_rotated = np.dot(rotation_matrix, v)  # Multiply the rotation matrix by the vector
+            return v_rotated
+
+        # Rotate the drag force direction to normalize it to 0 degrees
+        if orientation != 0:
+            drag_force_direction = rotate_vector_clockwise(drag_force_direction, orientation)
+
+        if (drag_force_direction[0] > 0 and drag_force_direction[1] > 0) or (
+            drag_force_direction[0] < 0 and drag_force_direction[1] < 0
+        ):
+            lift_force_direction = np.array([-drag_force_direction[1], drag_force_direction[0]])
+        elif (drag_force_direction[0] > 0 and drag_force_direction[1] < 0) or (
+            drag_force_direction[0] < 0 and drag_force_direction[1] > 0
+        ):
+            lift_force_direction = np.array([drag_force_direction[1], -drag_force_direction[0]])
+        else:
+            lift_force_direction = np.array([0, 0])
```

Comment:
Good catch on this corner case. If the drag force happens to align on an axis, the lifting force goes to zero.